airbot_sensor_to_pointcloud:
  ros__parameters:
  
    target_frame: "map" # "map" or "base_link"

    tof:
      all:
        use: true
      1D:
        use: true
        publish_rate_ms: 10
        tilting_angle_deg: 45.0 # double type
      multi:
        publish_rate_ms: 50
        left:
          use: true
        right:
          use: true
        row:
          use: true
          publish_rate_ms: 50
    
    camera:
      use: true
      publish_rate_ms: 100
      pointcloud_resolution: 0.05
      class_id_confidence_th: # 형식: "class id: confidence score"
        - "2: 55"
        - "5: 55"
        - "6: 55"
      object_direction: false  # 정방향(CCW+):True, 역방향(CW+):False
      logger:
        use: true
        margin:
          distance_diff: 0.5
          width_diff: 0.1
          height_diff: 0.1
    
    cliff:
      use: true
      publish_rate_ms: 10
  

    flip1: false
    flip2: false
    inverse1: false
    inverse2: false
    laser1Alpha: 180.0
    laser1AngleMax: 270.0  # 조정된 각도 범위
    laser1AngleMin: 90.0  # 조정된 각도 범위
    laser1B: 0
    laser1G: 0
    laser1R: 255
    laser1XOff: 0.15  # 앞쪽 라이다의 X 오프셋 (앞으로 0.05m)
    laser1YOff: 0.0
    laser1ZOff: 0.0
    laser2Alpha: 0.0
    laser2AngleMax: 270.0  # 조정된 각도 범위
    laser2AngleMin: 90.0  # 조정된 각도 범위
    laser2B: 255
    laser2G: 0
    laser2R: 0
    laser2XOff: -0.15  # 뒤쪽 라이다의 X 오프셋 (뒤로 0.05m)
    laser2YOff: 0.0
    laser2ZOff: 0.0
    pointCloudTopic: sensor_to_pointcloud/lidar
    pointCloutFrameId: base_scan
    qos_overrides./parameter_events.publisher.depth: 1000
    qos_overrides./parameter_events.publisher.reliability: best_effort #best_effort
    scanTopic1: /scan_front
    scanTopic2: /scan_back
    show1: true
    show2: true
    use_sim_time: false #false
